modules/claims_api/app/controllers/claims_api/v2/veterans/evidence_waiver_controller.rb

```diff
@@ -13,8 +18,14 @@ def submit
             raise ::Common::Exceptions::ResourceNotFound.new(detail: 'Claim not found')
           end
 
+          unless bgs_claim&.dig&.[](:filed_5103_waiver_ind) == FILE_5103
```

Comment:
we can just use .dig here. The `&.[](:filed5103_waver_ind)` would never actually be hit, it would always be nil. We have to dig in to `bnft_claim_dto` first... plus if any of this is nil it will nil.
```suggestion
          unless bgs_claim.dig(:bnft_claim_dto, :filed5103_waiver_ind) == FILE_5103
```